OSSM-8338: Update gateway content (#84361)
openshift/openshift-docs: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc

> $ oc label namespace httpbin istio-injection=enabled

Reviewer: Out of scope and totally up to you if you're game for content improvements, but these first two steps could use the addition of "by running the following command:"

Author: done

----

> . Deploy a sample service named `httpbin` by applying the `http.yaml` file: 

Reviewer 2: ```suggestion
. Deploy a sample service named `httpbin` by applying the `httpbin.yaml` file: 
```

Author: done

> . Deploy a sample service named `httpbin` by applying the `httpbin.yaml` file: 

Reviewer: This also needs that command language, but already has "by [doing some action]" wording. I'll leave it up to you to figure out, but maybe this could work:
```suggestion
. Apply the `httpbin.yaml` file to deploy a sample service named `httpbin` by running the following command: 
```
Not sure I'm satisfied with that, but you get the point and I trust that you know what's right. Maybe one of the "apply the yaml file" or "deploy a service" phrases can be dropped and still have the step be reasonably informative. 

Or maybe you can say "this deploys a service named X" at the end, even though we generally should avoid describing the results of steps like that

Author: done

+
[source,terminal]
----
$ oc apply -n httpbin -f https://raw.githubusercontent.com/openshift-service-mesh/istio/refs/heads/master/samples/httpbin/httpbin.yaml

> ----

Author: done

> $ oc create namespace curl

Reviewer: The wording of this step could use the same language if you want

----

> . Deploy the `curl` client by applying the `curl.yaml` file:

Reviewer: Same situation as earlier with this command. IMO, now that I'm reading a second command just like this, I feel like you can drop the "by applying the X yaml file" phrase from both of these steps. 

It seems like the name of the YAML file is not relevant given that it's only part of the filepaths in these commands, and users otherwise might not need to know about the file names more than the action they're taking

Author: done

+
[source,terminal]
----
$ oc apply -n curl -f https://raw.githubusercontent.com/openshift-service-mesh/istio/refs/heads/master/samples/curl/curl.yaml
----

> . Set a `CURL_POD` variable with the name of the curl pod:

Reviewer: ```suggestion
. Set a `CURL_POD` variable with the name of the curl pod by running the following command:
```

+
[source,terminal]
----
$ CURL_POD=$(oc get pods -n curl -l app=curl -o jsonpath='{.items[*].metadata.name}')
----

> . Using the `curl` client, send a request to the `/headers` endpoint of the `httpbin` application through the ingress gateway `Service` resource. Set the `Host` header of the request to `httpbin.example.com` to match the host that the {istio} `Gateway` and `VirtualService` resources specify.

Reviewer: This line should also mention the command language somewhere in the first sentence, though how you do that is up to you. Idk enough about the context of this to suggest what to move around or leave out

Reviewer: Also, at this point I'll stop making this comment and just ask that you add this language at least to any applicable step you touch in your PR.

Author: done

+

> ----
> [source,terminal]

Reviewer: These lines need to be switched so that the source block isn't in the actual command syntax:
```suggestion
[source,terminal]
----
```

+
[source,terminal]
----
$ oc apply -n curl -f https://raw.githubusercontent.com/openshift-service-mesh/istio/refs/heads/master/samples/curl/curl.yaml

> ----

Author: done

A gateway can also be exposed to traffic outside the cluster by using {ocp-short-name} Routes. For more information, see "Exposing a gateway to traffic outside the cluster using {ocp-short-name} Routes".
====

> . Verify that `httpbin` service can be accessed from outside the cluster through the external hostname or IP address of the gateway `Service` resource. Ensure that you set the `INGRESS_HOST` variable appropriately for the environment that your cluster is running in. 

Author: done

+
.. If the cluster runs on AWS, run the following command to set the `INGRESS_HOST` variable:
[source,terminal]
----
$ INGRESS_HOST=$(oc get service <gateway_name> -n <gateway_namespace> -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
----
+
.. If the cluster runs on GCP or Azure, run the following command to set the `INGRESS_HOST` variable:
[source,terminal]
----
$ INGRESS_HOST=$(oc get service <gateway_name> -n <gateway_namespace> -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
----
+ 
.. Send a `curl` request to the `httpbin` service using the host of the gateway:
[source,terminal]
----
$ curl -s -I -H Host:httpbin.example.com http://$INGRESS_HOST/headers 
----
+

> Verify that the response has the `HTTP/1.1 200 OK` status, which indicates that the request was successful.

Reviewer 2: The same formatting issue mentioned earlier is also present here.
